File: output/deepseek_r1/codenet/algo_based_translation/java/python/s527301188.py

strs = input().split()
nums = [0] * len(strs)
top = 0

for token in strs:
    if token in '+-*':
        top -= 1
        if token == '+':
            nums[top - 1] += nums[top]
        elif token == '-':
            nums[top - 1] -= nums[top]
        elif token == '*':
            nums[top - 1] *= nums[top]
    else:
        nums[top] = int(token)
        top += 1

print(nums[0])

File: dataset/codenet/Java/Code/s527301188.java

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Arrays;

public class Main {

	public static void main(String[] args) throws NumberFormatException,
			IOException {
		BufferedReader reader = new BufferedReader(new InputStreamReader(
				System.in));

		String[] strs = reader.readLine().split(" ");
		int[] nums = new int[strs.length];
		int top = 0;
		for (int i = 0; i < strs.length; i++) {
			switch (strs[i].charAt(0)) {
			case '+':
				top--;
				nums[top-1] = nums[top - 1] + nums[top];
				break;
			case '-':
				top--;
				nums[top-1] = nums[top - 1] - nums[top];
				break;
			case '*':
				top--;
				nums[top-1] = nums[top - 1] * nums[top];
				break;
			default:
				
				nums[top] = Integer.parseInt(strs[i]);
				top++;
				break;
			}
			/*System.out.println(Arrays.toString(nums));*/
		}
		System.out.println(nums[0]);

	}
}